### PR TITLE
implement SetEndOfFile, CreateDirectoryA, RemoveDirectoryA, SetFileAttributesA

### PR DESCRIPTION
**Pull request by camthesaxman**
These functions are needed for mwldeppc.exe to work.
Note that these implementations are rather dummy in nature and not fully compliant to the Windows API.

### CODE REVIEW

File: kernel32.cpp [hunk 1]
```diff
@@ -437,6 +437,30 @@ namespace kernel32 {
 		return r;
 	}
 
+	int WIN_FUNC SetEndOfFile(void *hFile) {
+		DEBUG_LOG("SetEndOfFile\n");
+		FILE *fp = files::fpFromHandle(hFile);
+		return ftruncate(fileno(fp), ftell(fp)) == 0;
```
Comment (simonlindholm) [inline kernel32.cpp:443]: fflush(fp) before to make sure there's nothing cached?

**Review by simonlindholm — APPROVED** [review]

File: kernel32.cpp (hunk 1 above)
Comment (camthesaxman) [reply kernel32.cpp:443]: Is it necessary to call flush first? I thought it would happen automatically since the file API is supposed to be synchronous.

Comment (simonlindholm) [reply kernel32.cpp:443]: I believe so. https://stackoverflow.com/questions/2806104/does-fwrite-buffer-the-output

Comment (camthesaxman) [reply kernel32.cpp:443]: Well, that's for fwrite, not ftruncate, but I'll do the flush just to be sure.

Comment (simonlindholm) [reply kernel32.cpp:444]: The point is that fwrite may buffer, while ftruncate operates on file descriptors which is the layer below file handles, so it won't flush those buffers before running. `ftell` may, but I wouldn't rely on that.

**Pull request merged**